pkg/splunk/enterprise/configuration.go

```diff
@@ -675,29 +675,33 @@ func updateSplunkPodTemplateWithConfig(client splcommon.ControllerClient, podTem
 		clusterMasterURL = "localhost"
 	} else if spec.ClusterMasterRef.Name != "" {
 		clusterMasterURL = GetSplunkServiceName(SplunkClusterMaster, spec.ClusterMasterRef.Name, false)
-		if spec.ClusterMasterRef.Namespace != "" {
-			clusterMasterURL = splcommon.GetServiceFQDN(spec.ClusterMasterRef.Namespace, clusterMasterURL)
-		}
-		//Check if CM is connected to a LicenseMaster
-		namespacedName := types.NamespacedName{
-			Namespace: cr.GetNamespace(),
-			Name:      spec.ClusterMasterRef.Name,
-		}
-		masterIdxCluster := &enterprisev1.ClusterMaster{}
-		err := client.Get(context.TODO(), namespacedName, masterIdxCluster)
-		if err != nil {
-			scopedLog.Error(err, "Unable to get ClusterMaster")
-		}
 
-		if masterIdxCluster.Spec.LicenseMasterRef.Name != "" {
-			licenseMasterURL := GetSplunkServiceName(SplunkLicenseMaster, masterIdxCluster.Spec.LicenseMasterRef.Name, false)
-			if masterIdxCluster.Spec.LicenseMasterRef.Namespace != "" {
-				licenseMasterURL = splcommon.GetServiceFQDN(masterIdxCluster.Spec.LicenseMasterRef.Namespace, licenseMasterURL)
+		// Avoid for LM
+		if instanceType != SplunkLicenseMaster {
```

Comment:
I am sorry but I have a trivial question here. If the instanceType is SplunkLicenseMaster, then how would it pick up the clusterMasterURL?

Reply:
Line 677. 

`		clusterMasterURL = GetSplunkServiceName(SplunkClusterMaster, spec.ClusterMasterRef.Name, false)
`

Followed by line 708

```
	if clusterMasterURL != "" {
		extraEnv = append(extraEnv, corev1.EnvVar{
			Name:  "SPLUNK_CLUSTER_MASTER_URL",
			Value: clusterMasterURL,
		})
	}
```

Reply:
oh damn, I missed it completely